Control what perception publishes tag to

qhdwight opened the pull request:
## Summary
Closes #369 

Currently perception hard codes the frame it publishes tags to. Read `use_odom` ros param to decide whether to publish to odometry or map frame.

### Did you add documentation to the wiki?
No

## How was this code tested? 
Ran auton sim, went up to post with teleop twist, and made sure detections were present

### Did you test this in sim? 
Yes

### Did you test this on the rover?
No

rbridges12 commented:
looks good, lmk when its tested and I'll approve

ankithu reviewed (commented):
yea looks good, same as riley

qhdwight commented:
Tested in sim works fine. I also made a few other QOL changes:
- Better lighting in gazebo so tag detections happen more often (still need to look into why so poor)
- Renamed fiducial_images to tag_detection
- Fixed annoying spam issue with kinect plugin and tag detector node

rbridges12 commented:
so you tested it for publishing to odom and map frame?

Also one thing you could do to make sim detections better is have a separate set of detection params that only gets used in sim, kinda like im doing with the ekf

niwhsa9 reviewed (approved):
> constexpr size_t IMAGE_WIDTH_WARN_SIZE = 640;

lol sweaty cuz ur only using it at runtime  :smile:

rbridges12 reviewed (approved):
lgtm
